src/fiskaltrust.Middleware.Interface.Http/HttpPos.cs

```diff
@@ -0,0 +1,227 @@
+﻿using fiskaltrust.ifPOS.v1;
+using fiskaltrust.Middleware.Interface.Http.Helpers;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace fiskaltrust.Middleware.Interface.Http
+{
+    public class HttpPos : IPOS
+    {
+        private readonly string _url;
+        private readonly POSOptions _options;
+
+        private delegate string AsyncEchoCaller(string message);
+        private delegate ifPOS.v0.ReceiptResponse AsyncSignCaller(ifPOS.v0.ReceiptRequest request);
+        private delegate Stream AsyncJournalCaller(long ftJournalType, long from, long to);
+
+        public HttpPos(POSOptions options)
+        {
+            _url = options.Url.Replace("rest://", "http://").Replace("xml://", "http://");
```

Comment:
That would mean that if you copy the URL that is printed by the middleware, this will fail. But i get your point, not sure what makes more sense here..